model/roles.go

```diff
@@ -400,30 +400,28 @@ func (r *Role) GetScriptPaths() map[string]string {
 func (r *Role) GetScriptSignatures() (string, error) {
 	hasher := sha1.New()
 
-	i := 0
 	paths := r.GetScriptPaths()
-	scripts := make([]string, len(paths))
+	scripts := make([]string, 0, len(paths))
 
-	for _, f := range paths {
-		scripts[i] = f
-		i++
+	for filename := range paths {
+		scripts = append(scripts, filename)
 	}
 
 	sort.Strings(scripts)
 
 	for _, filename := range scripts {
 		hasher.Write([]byte(filename))
 
-		f, err := os.Open(filename)
+		f, err := os.Open(paths[filename])
 		if err != nil {
 			return "", err
 		}
 
-		if _, err := io.Copy(hasher, f); err != nil {
+		_, err = io.Copy(hasher, f)
+		f.Close()
```

Comment:
We could, but (exactly due to the looping) I thought it would be better to avoid doing it.  I actually wrote that first 😄